### ISSUE TITLE
Error on load using 0.17.2: `java.lang.NoClassDefFoundError: org/apache/spark/sql/sources/v2/DataSourceOptions`

### ISSUE THREAD

**Issue by acmiyaguchi** [GoogleCloudDataproc/spark-bigquery-connector#241]
`gs://spark-lib/bigquery/spark-bigquery-latest_2.12.jar` corresponding to `spark-bigquery-with-dependencies_2.12-0.17.2.jar` breaks a job running on Dataproc using the `preview-ubuntu18` image, with the following message:

```
py4j.protocol.Py4JJavaError: An error occurred while calling o55.load.
: java.lang.NoClassDefFoundError: org/apache/spark/sql/sources/v2/DataSourceOptions
	at com.google.cloud.spark.bigquery.SparkBigQueryConfig.fromV1(SparkBigQueryConfig.java:117)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.createSparkBigQueryConfig(BigQueryRelationProvider.scala:138)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.createRelationInternal(BigQueryRelationProvider.scala:70)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.createRelation(BigQueryRelationProvider.scala:44)
	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:340)
	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:279)
	at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:268)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:268)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:203)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: org.apache.spark.sql.sources.v2.DataSourceOptions
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 21 more
```

Pinning the library to `gs://spark-lib/bigquery/spark-bigquery-with-dependencies_2.12-0.17.1.jar` fixes the issue.

**Comment by medb** [comment]
Thank you for the report, we rolled back `-latest` links to point to `0.17.1` connector version to mitigate the issue.

**Comment by davidrabinowitz** [comment]
Fixed by PR #242
